From 40f646e369dece1bccfdcf2d371301899d1b38d5 Mon Sep 17 00:00:00 2001
From: Luo Jie <quic_luoj@quicinc.com>
Date: Tue, 2 Apr 2024 13:52:51 +0800
Subject: [PATCH 512/512] arm64: dts: Add RDP453 DTS

RDP453 is connected with QCA8084 PHY via PCS0, connected with
two AQ PHYs via PCS1 and PCS2.

Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
---
 arch/arm64/boot/dts/qcom/ipq9574-rdp453.dts | 438 ++++++++++++++++++++
 1 file changed, 438 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/ipq9574-rdp453.dts b/arch/arm64/boot/dts/qcom/ipq9574-rdp453.dts
index c30c9fbedf26..245e6783c020 100644
--- a/arch/arm64/boot/dts/qcom/ipq9574-rdp453.dts
+++ b/arch/arm64/boot/dts/qcom/ipq9574-rdp453.dts
@@ -10,8 +10,446 @@
 
 #include "ipq9574-rdp-common.dtsi"
 
+#include <dt-bindings/clock/qcom,qca8k-nsscc.h>
+#include <dt-bindings/net/qcom,qca808x.h>
+#include <dt-bindings/reset/qcom,qca8k-nsscc.h>
+
 / {
 	model = "Qualcomm Technologies, Inc. IPQ9574/AP-AL02-C8";
 	compatible = "qcom,ipq9574-ap-al02-c8", "qcom,ipq9574";
 
+	clocks {
+		qca8k_uniphy1_rx312p5m: qca8k-uniphy1-rx312p5m {
+			compatible = "fixed-clock";
+			clock-frequency = <312500000>;
+			#clock-cells = <0>;
+		};
+
+		qca8k_uniphy1_tx312p5m: qca8k-uniphy1-tx312p5m {
+			compatible = "fixed-clock";
+			clock-frequency = <312500000>;
+			#clock-cells = <0>;
+		};
+	};
+};
+
+&pcs0 {
+        status = "okay";
+};
+
+&pcs0_ch0 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT1_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT1_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs0_ch1 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT2_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT2_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs0_ch2 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT3_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT3_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs0_ch3 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT4_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT4_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs1 {
+        status = "okay";
+};
+
+&pcs1_ch0 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT5_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT5_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&pcs2{
+        status = "okay";
+};
+
+&pcs2_ch0 {
+        clocks = <&nsscc NSS_CC_UNIPHY_PORT6_RX_CLK>,
+                 <&nsscc NSS_CC_UNIPHY_PORT6_TX_CLK>;
+        clock-names = "rx",
+                      "tx";
+        status = "okay";
+};
+
+&mdio {
+	clock-frequency = <6250000>;
+	status = "okay";
+
+	phy5: ethernet-phy@0 {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <0>;
+	};
+
+        ethernet-phy-package@1 {
+            #address-cells = <1>;
+            #size-cells = <0>;
+            compatible = "qcom,qca8084-package";
+            reg = <1>;
+            clocks = <&qca8k_nsscc NSS_CC_APB_BRIDGE_CLK>,
+                     <&qca8k_nsscc NSS_CC_AHB_CLK>,
+                     <&qca8k_nsscc NSS_CC_SEC_CTRL_AHB_CLK>,
+                     <&qca8k_nsscc NSS_CC_TLMM_CLK>,
+                     <&qca8k_nsscc NSS_CC_TLMM_AHB_CLK>,
+                     <&qca8k_nsscc NSS_CC_CNOC_AHB_CLK>,
+                     <&qca8k_nsscc NSS_CC_MDIO_AHB_CLK>;
+            clock-names = "apb_bridge",
+                          "ahb",
+                          "sec_ctrl_ahb",
+                          "tlmm",
+                          "tlmm_ahb",
+                          "cnoc_ahb",
+                          "mdio_ahb";
+            qcom,package-mode = <QCA808X_PCS1_10G_QXGMII_PCS0_UNUNSED>;
+            resets = <&qca8k_nsscc NSS_CC_GEPHY_FULL_ARES>;
+            qcom,phy-addr-fixup = <1 2 3 4 5 6 7>;
+
+	    phy0: ethernet-phy@1 {
+                compatible = "ethernet-phy-id004d.d180";
+                reg = <1>;
+                clocks = <&qca8k_nsscc NSS_CC_GEPHY0_SYS_CLK>;
+                resets = <&qca8k_nsscc NSS_CC_GEPHY0_SYS_ARES>;
+                qcom,xpcs-channel = <0>;
+            };
+
+	    phy1: ethernet-phy@2 {
+                compatible = "ethernet-phy-id004d.d180";
+                reg = <2>;
+                clocks = <&qca8k_nsscc NSS_CC_GEPHY1_SYS_CLK>;
+                resets = <&qca8k_nsscc NSS_CC_GEPHY1_SYS_ARES>;
+                qcom,xpcs-channel = <1>;
+            };
+
+	    phy2: ethernet-phy@3 {
+                compatible = "ethernet-phy-id004d.d180";
+                reg = <3>;
+                clocks = <&qca8k_nsscc NSS_CC_GEPHY2_SYS_CLK>;
+                resets = <&qca8k_nsscc NSS_CC_GEPHY2_SYS_ARES>;
+                qcom,xpcs-channel = <2>;
+            };
+
+	    phy3: ethernet-phy@4 {
+                compatible = "ethernet-phy-id004d.d180";
+                reg = <4>;
+                clocks = <&qca8k_nsscc NSS_CC_GEPHY3_SYS_CLK>;
+                resets = <&qca8k_nsscc NSS_CC_GEPHY3_SYS_ARES>;
+                qcom,xpcs-channel = <3>;
+            };
+
+            pcs-phy@6 {
+                compatible = "qcom,qca8k-pcs-phy";
+                reg = <6>;
+                clocks = <&qca8k_nsscc NSS_CC_SRDS1_SYS_CLK>,
+                         <&qca8k_uniphy1_tx312p5m>,
+                         <&qca8k_uniphy1_rx312p5m>;
+                clock-names = "pcs", "pcs_rx_root", "pcs_tx_root";
+                resets = <&qca8k_nsscc NSS_CC_SRDS1_SYS_ARES>;
+            };
+
+            xpcs-phy@7 {
+                compatible = "qcom,qca8k-xpcs-phy";
+                reg = <7>;
+                #address-cells = <1>;
+                #size-cells = <0>;
+                resets = <&qca8k_nsscc NSS_CC_XPCS_ARES>;
+
+                channel@0 {
+                    reg = <0>;
+                    clocks = <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_XGMII_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_XGMII_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC1_GEPHY0_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC1_GEPHY0_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC1_RX_CLK_SRC>,
+                             <&qca8k_nsscc NSS_CC_MAC1_TX_CLK_SRC>;
+                    clock-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx",
+                                  "rx_src",
+                                  "tx_src";
+                    resets = <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_XGMII_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_XGMII_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC1_SRDS1_CH0_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC1_GEPHY0_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC1_GEPHY0_TX_ARES>;
+                    reset-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx";
+                };
+
+                channel@1 {
+                    reg = <1>;
+                    clocks = <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_XGMII_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_XGMII_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC2_GEPHY1_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC2_GEPHY1_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC2_RX_CLK_SRC>,
+                             <&qca8k_nsscc NSS_CC_MAC2_TX_CLK_SRC>;
+                    clock-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx",
+                                  "rx_src",
+                                  "tx_src";
+                    resets = <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_XGMII_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_XGMII_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC2_SRDS1_CH1_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC2_GEPHY1_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC2_GEPHY1_TX_ARES>;
+                    reset-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx";
+                };
+
+                channel@2 {
+                    reg = <2>;
+                    clocks = <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_XGMII_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_XGMII_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC3_GEPHY2_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC3_GEPHY2_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC3_RX_CLK_SRC>,
+                             <&qca8k_nsscc NSS_CC_MAC3_TX_CLK_SRC>;
+                    clock-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx",
+                                  "rx_src",
+                                  "tx_src";
+                    resets = <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_XGMII_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_XGMII_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC3_SRDS1_CH2_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC3_GEPHY2_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC3_GEPHY2_TX_ARES>;
+                    reset-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx";
+                };
+
+                channel@3 {
+                    reg = <3>;
+                    clocks = <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_XGMII_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_XGMII_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC4_GEPHY3_RX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC4_GEPHY3_TX_CLK>,
+                             <&qca8k_nsscc NSS_CC_MAC4_RX_CLK_SRC>,
+                             <&qca8k_nsscc NSS_CC_MAC4_TX_CLK_SRC>;
+                    clock-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx",
+                                  "rx_src",
+                                  "tx_src";
+                    resets = <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_XGMII_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_XGMII_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_TX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC4_SRDS1_CH3_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC4_GEPHY3_RX_ARES>,
+                             <&qca8k_nsscc NSS_CC_MAC4_GEPHY3_TX_ARES>;
+                    reset-names = "xgmii_rx",
+                                  "xgmii_tx",
+                                  "xpcs_rx",
+                                  "xpcs_tx",
+                                  "port_rx",
+                                  "port_tx";
+                };
+            };
+        };
+
+	phy4: ethernet-phy@8 {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <8>;
+	};
+
+	qca8k_nsscc: qca8k-nsscc@18 {
+		compatible = "qcom,qca8084-nsscc";
+		reg = <0x18>;
+		#clock-cells = <1>;
+		#reset-cells = <1>;
+		reset-gpios = <&tlmm 60 GPIO_ACTIVE_LOW>;
+		clocks = <&cmn_pll ETH0_50MHZ_CLK>,
+			 <0>,
+			 <0>,
+			 <0>,
+			 <0>,
+			 <&qca8k_uniphy1_rx312p5m>,
+			 <&qca8k_uniphy1_tx312p5m>;
+	};
+};
+
+&qcom_ppe {
+	ethernet-ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@1 {
+			reg = <1>;
+			phy-mode = "10g-qxgmii";
+			label = "lan1";
+			phy-handle = <&phy0>;
+			pcs-handle = <&pcs0_ch0>;
+			clocks = <&nsscc NSS_CC_PORT1_MAC_CLK>,
+				 <&nsscc NSS_CC_PORT1_RX_CLK>,
+				 <&nsscc NSS_CC_PORT1_TX_CLK>;
+			clock-names = "mac",
+				      "rx",
+				      "tx";
+			resets = <&nsscc PORT1_MAC_ARES>,
+				 <&nsscc PORT1_RX_ARES>,
+				 <&nsscc PORT1_TX_ARES>;
+			reset-names = "mac",
+				      "rx",
+				      "tx";
+		};
+
+		port@2 {
+			reg = <2>;
+			phy-mode = "10g-qxgmii";
+			label = "lan2";
+			phy-handle = <&phy1>;
+			pcs-handle = <&pcs0_ch1>;
+			clocks = <&nsscc NSS_CC_PORT2_MAC_CLK>,
+				 <&nsscc NSS_CC_PORT2_RX_CLK>,
+				 <&nsscc NSS_CC_PORT2_TX_CLK>;
+			clock-names = "mac",
+				      "rx",
+				      "tx";
+			resets = <&nsscc PORT2_MAC_ARES>,
+				 <&nsscc PORT2_RX_ARES>,
+				 <&nsscc PORT2_TX_ARES>;
+			reset-names = "mac",
+				      "rx",
+				      "tx";
+		};
+
+		port@3 {
+			reg = <3>;
+			phy-mode = "10g-qxgmii";
+			label = "lan3";
+			phy-handle = <&phy2>;
+			pcs-handle = <&pcs0_ch2>;
+			clocks = <&nsscc NSS_CC_PORT3_MAC_CLK>,
+				 <&nsscc NSS_CC_PORT3_RX_CLK>,
+				 <&nsscc NSS_CC_PORT3_TX_CLK>;
+			clock-names = "mac",
+				      "rx",
+				      "tx";
+			resets = <&nsscc PORT3_MAC_ARES>,
+				 <&nsscc PORT3_RX_ARES>,
+				 <&nsscc PORT3_TX_ARES>;
+			reset-names = "mac",
+				      "rx",
+				      "tx";
+		};
+
+		port@4 {
+			reg = <4>;
+			phy-mode = "10g-qxgmii";
+			label = "lan4";
+			phy-handle = <&phy3>;
+			pcs-handle = <&pcs0_ch3>;
+			clocks = <&nsscc NSS_CC_PORT4_MAC_CLK>,
+				 <&nsscc NSS_CC_PORT4_RX_CLK>,
+				 <&nsscc NSS_CC_PORT4_TX_CLK>;
+			clock-names = "mac",
+				      "rx",
+				      "tx";
+			resets = <&nsscc PORT4_MAC_ARES>,
+				 <&nsscc PORT4_RX_ARES>,
+				 <&nsscc PORT4_TX_ARES>;
+			reset-names = "mac",
+				      "rx",
+				      "tx";
+		};
+
+		port@5 {
+			reg = <5>;
+			phy-mode = "usxgmii";
+			managed = "in-band-status";
+			label = "lan5";
+			phy-handle = <&phy4>;
+			pcs-handle = <&pcs1_ch0>;
+			clocks = <&nsscc NSS_CC_PORT5_MAC_CLK>,
+				 <&nsscc NSS_CC_PORT5_RX_CLK>,
+				 <&nsscc NSS_CC_PORT5_TX_CLK>;
+			clock-names = "mac",
+				      "rx",
+				      "tx";
+			resets = <&nsscc PORT5_MAC_ARES>,
+				 <&nsscc PORT5_RX_ARES>,
+				 <&nsscc PORT5_TX_ARES>;
+			reset-names = "mac",
+				      "rx",
+				      "tx";
+		};
+
+		port@6 {
+			reg = <6>;
+			phy-mode = "usxgmii";
+			managed = "in-band-status";
+			label = "wan";
+			phy-handle = <&phy5>;
+			pcs-handle = <&pcs2_ch0>;
+			clocks = <&nsscc NSS_CC_PORT6_MAC_CLK>,
+				 <&nsscc NSS_CC_PORT6_RX_CLK>,
+				 <&nsscc NSS_CC_PORT6_TX_CLK>;
+			clock-names = "mac",
+				      "rx",
+				      "tx";
+			resets = <&nsscc PORT6_MAC_ARES>,
+				 <&nsscc PORT6_RX_ARES>,
+				 <&nsscc PORT6_TX_ARES>;
+			reset-names = "mac",
+				      "rx",
+				      "tx";
+		};
+	};
 };
-- 
2.34.1

